ApplicationLibCode/Commands/RicImportEnsembleSurfaceFeature.cpp

```diff
@@ -75,42 +76,79 @@ void RicImportEnsembleSurfaceFeature::importEnsembleSurfaceFromFiles( const QStr
 {
     if ( fileNames.isEmpty() ) return;
 
-    QString ensembleName = RiaEnsembleNameTools::findSuitableEnsembleName( fileNames );
-    QString layerName    = RiaEnsembleNameTools::findCommonBaseName( fileNames );
-    if ( !layerName.isEmpty() )
+    // Create a list of file names for each layer
+    std::map<QString, QStringList> fileNamesForEachLayer;
     {
-        ensembleName += QString( " : %1" ).arg( layerName );
-    }
+        for ( const auto& name : fileNames )
+        {
+            QFileInfo fi( name );
 
-    if ( ensembleName.isEmpty() ) ensembleName = "Ensemble Surface";
+            auto layerName = fi.baseName();
+            fileNamesForEachLayer[layerName].push_back( name );
+        }
+    }
 
-    std::vector<RimFileSurface*> surfaces;
-    for ( const auto& fileName : fileNames )
+    RimEnsembleSurface* ensembleToSelect = nullptr;
+    for ( const auto& [layer, fileNames] : fileNamesForEachLayer )
     {
-        RimFileSurface* fileSurface = new RimFileSurface;
-        fileSurface->setSurfaceFilePath( fileName );
+        QString ensembleName = RiaEnsembleNameTools::findSuitableEnsembleName( fileNames );
+        QString layerName    = layer;
+        if ( !layerName.isEmpty() )
+        {
+            ensembleName += QString( " : %1" ).arg( layerName );
+        }
 
-        auto shortName = RiaEnsembleNameTools::uniqueShortName( fileName, fileNames );
-        fileSurface->setUserDescription( shortName );
+        if ( ensembleName.isEmpty() ) ensembleName = "Ensemble Surface";
 
-        if ( fileSurface->onLoadData() )
+        std::map<QString, QStringList> keyFileComponentsForAllFiles =
+            RiaFilePathTools::keyPathComponentsForEachFilePath( fileNames );
+
+        std::vector<RimFileSurface*> surfaces;
+
+        int fileCount = static_cast<int>( fileNames.size() );
+#pragma omp parallel for
+        for ( int i = 0; i < fileCount; i++ )
         {
-            surfaces.push_back( fileSurface );
+            auto fileName = fileNames[i];
+
+            RimFileSurface* fileSurface = nullptr;
+#pragma omp critical( new_rimFileSurface ) // Creation of a new Pdm object is not thread-safe
+            fileSurface = new RimFileSurface;
+
+            fileSurface->setSurfaceFilePath( fileName );
+
+            auto shortName =
+                RiaEnsembleNameTools::uniqueShortNameFromComponents( fileName, keyFileComponentsForAllFiles, ensembleName );
+            fileSurface->setUserDescription( shortName );
```

Comment:
Maybe create the RimFileSurface in the other critical region (the names can be computed in paralell)?